### PR TITLE
Create babel-plugin-transform-conditional-imports for babel transform envs

### PR DESCRIPTION
**Pull request by JakeLane**
## Motivation

Some environments (like webpack and jest) need to transform `importCond` without errors. This change is similar to what's in feature flag off state for conditional bundling.

## Changes

New babel plugin for use in environments where atlaspack is not transforming code.

## Checklist

- [x] Existing or new tests cover this change

### CODE REVIEW

**Comment by JakeLane** [comment]
I ended up implementing a swc plugin that can be used externally. Unfortunately, caused some issues when trying to slot into swc-loader and jest, so I've had to drop this. Given we reuse the code here we can just migrate over in the future and delete the babel plugin.

File: packages/utils/babel-plugin-transform-conditional-imports/test/babel-plugin-transform-conditional-imports.test.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    assert(
+      transformed ===
+        `const _CONDITION$IF_TRUE$IF_FALSE = {
+  ifTrue: require('IF_TRUE').default,
+  ifFalse: require('IF_FALSE').default
+};
+Object.defineProperty(_CONDITION$IF_TRUE$IF_FALSE, "load", {
+  get: () => globalThis.__MCOND && globalThis.__MCOND('CONDITION') ? _CONDITION$IF_TRUE$IF_FALSE.ifTrue : _CONDITION$IF_TRUE$IF_FALSE.ifFalse
+});
+_CONDITION$IF_TRUE$IF_FALSE.load;`,
```
Comment (tomgasson) [inline packages/utils/babel-plugin-transform-conditional-imports/test/babel-plugin-transform-conditional-imports.test.js:44]: I'm not sure I get this - it's immediately called?

File: crates/atlaspack_conditional_imports/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,346 @@
...
+    &mut self,
+    cond: Atom,
+    if_true: Atom,
+    if_true_span: Span,
+    if_false: Atom,
+    if_false_span: Span,
+  ) -> Expr {
+    // importCond('CONDITION', 'IF_TRUE', 'IF_FALSE');
+    // =>
+    // (globalThis.__MCOND && globalThis.__MCOND['CONDITION'] ? require('IF_TRUE') : require('IF_FALSE')).default;
```
Comment (tomgasson) [inline crates/atlaspack_conditional_imports/src/lib.rs:81]: Can we just hard fail if `globalThis.__MCOND` isn't defined? I don't want this going into unknown places... better we actually set them up

File: packages/utils/babel-plugin-transform-conditional-imports/test/babel-plugin-transform-conditional-imports.test.js (hunk 1 above)
Comment (JakeLane) [reply packages/utils/babel-plugin-transform-conditional-imports/test/babel-plugin-transform-conditional-imports.test.js:44]: `importCond` isn't assigned to a variable so it just becomes a lone identifier in this state of the pipeline

File: packages/transformers/js/core/src/lib.rs [hunk 3]
```diff
@@ -359,7 +357,18 @@ pub fn transform(
 
               // Replace conditional imports with requires when flag is off
               if !config.conditional_bundling {
-                module.visit_mut_with(&mut ConditionalImportsFallback { unresolved_mark });
+                module.visit_mut_with(
```
Comment (tomgasson) [inline packages/transformers/js/core/src/lib.rs:360]: This looks like it's the wrong way around.  !conditional_bundling, yet we're visiting it with the conditional imports visitor?

Comment (JakeLane) [reply packages/transformers/js/core/src/lib.rs:360]: This visitor is used for the fallback without bundling behaviour. Perhaps I can make this clearer by renaming the visitor?

**Review by tomgasson — APPROVED** [review]

File: crates/atlaspack_swc_transform_utils/Cargo.toml [hunk 4]
```diff
@@ -0,0 +1,12 @@
+[package]
+name = "atlaspack_swc_transform_utils"
```
Comment (MonicaOlejniczak) [inline crates/atlaspack_swc_transform_utils/Cargo.toml:2]: Can this be named more specifically so we have more focused abstractions? Something like `atlaspack_swc_runner`

File: crates/atlaspack_swc_transform_utils/Cargo.toml [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+version = "0.1.0"
+edition = "2021"
+description = "Atlaspack Bundler"
```
Comment (MonicaOlejniczak) [inline crates/atlaspack_swc_transform_utils/Cargo.toml:5]: This should be filled out

File: package.json [hunk 6]
```diff
@@ -10,11 +10,12 @@
   "workspaces": [
     "crates/caniuse_database",
     "crates/lmdb-js-lite",
+    "crates/atlaspack_contextual_imports_swc_plugin",
```
Comment (MonicaOlejniczak) [inline package.json:13]: /nit make it alphabetical

File: packages/utils/babel-plugin-transform-contextual-imports/tsconfig.json [hunk 7]
```diff
@@ -0,0 +1,12 @@
...
+  "compilerOptions": {
+    "target": "es2016",
+    "module": "commonjs",
+    "esModuleInterop": true,
+    "forceConsistentCasingInFileNames": true,
+    "strict": true,
+    "skipLibCheck": true
+  },
+  "include": ["src/*"],
+  "exclude": ["node_modules", ".vscode-test"]
```
Comment (MonicaOlejniczak) [inline packages/utils/babel-plugin-transform-contextual-imports/tsconfig.json:11]: Is `.vscode-test` needed? Can you remove it?

Comment (JakeLane) [reply packages/utils/babel-plugin-transform-contextual-imports/tsconfig.json:11]: This is just duplicated from the other TS package (`packages/utils/atlaspackforvscode/tsconfig.json`). Makes sense to drop it

**Review by MonicaOlejniczak — APPROVED** [review]

**Pull request merged**